crypto/azure/keyvault/component.go

```diff
@@ -401,6 +399,14 @@ func (k *keyvaultCrypto) getVaultURI() string {
 	return fmt.Sprintf("https://%s.%s", k.md.VaultName, k.md.vaultDNSSuffix)
 }
 
+func (keyvaultCrypto) SupportedEncryptionAlgorithms() []string {
```

Comment:
should we let the user know that they have to call init before calling this method? 
```suggestion
func (keyvaultCrypto) SupportedEncryptionAlgorithms() ([]string, error) {
```
This way we can return error if this method is called without calling Init.

Reply:
Or do something similar to local base component, call algsParsed.Do() before returning the value. same below.

Reply:
I think having the assumption that `Init()` be called before this method is fair, and I don't think we need to make the method return an error.

The way this method is meant to be used is by the component metadata API, the same one that calls `Features()`: 
https://github.com/dapr/dapr/blob/263a21cf1a3d1bad5a4f261d1785fbe3eab09785/pkg/runtime/runtime.go#L3117